services/forgetPasswordService.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Created by shujaatali on 04/05/16.
+ */
+var config = require('../nodejs_config/config.js');
+var jwt = require('jsonwebtoken');
+var mailgun = require('mailgun-js')({apiKey: config.App.mailGunApiKey, domain: config.App.mailGunDomain});
+var db = require('nano-blue')(config.App.CouchServerIp+'/members');
+exports.checkUserName = function(req,res){
+
+
+        db.view('cbs', 'getMemberByUsername', {key: req.body.username, include_docs: true}, function(err, respBody) {
+        if (err) {
+        return err;
+             } else {
+            if (respBody.rows.length === 0) { // a doc with this 'username' does not exist in db
+            return 0;
```

Comment:
koi message ya error code bhi set karlo so that you can give user a helpful message like "the username you tried does not exist" etc.